### ISSUE TITLE
No Canadian major routes show up at middle-low zooms

### ISSUE THREAD

**Issue by andrepoiy** [ZeLonewolf/openstreetmap-americana#565]
So once you zoom out to a certain point, all Canadian roads disappear from the map, but it seems like some US Interstates still remain. 

I'm guessing that the Interstates have a special rule where they can be rendered at a lower zoom than everything else. In that case, could a similar rule be applied to major Canadian roads? 

Last zoom where Canadian roads are still visible
![image](https://user-images.githubusercontent.com/47615278/200139263-ad76bb75-3c57-42b5-884d-3c9fc89d484a.png)

Next zoom where all Canadian roads disappear
![image](https://user-images.githubusercontent.com/47615278/200139276-1d26fa94-4078-4659-9a5c-97536ab0f882.png)

**Comment by claysmalley** [comment]
> I'm guessing that the Interstates have a special rule where they can be rendered at a lower zoom than everything else.

Indeed, that is the case. Motorways are present in the vector tiles starting at zoom 4, and trunks at zoom 5. Instead of displaying all motorways at zoom 4, we decided to filter away any highways that aren't part of our national motorway network, but this filter goes away when all trunks are shown.

Of course, due to Canada's size and low population density, a national motorway network would be impractical. Though I personally believe the Trans-Canada Highway, among other non-motorway (inter)national networks, deserves the same highlighting at low zoom.

For this to happen, there would need to be a change to OpenMapTiles to display trunk roads at zoom 4.

**Comment by ZeLonewolf** [comment]
It's a valid issue, and I'd like to see route networks that are roughly equivalent to US interstates in other countries render at the same zoom level. I haven't tried to tackle this because I'm far less familiar with other countries' route networks.

One answer would be to promote all of `network=CA:transcanada` and `network=CA:transcanada:namedRoute` to be at a level equivalent to US interstates. I'm not sure what the difference is between the two network values other than the `namedRoute` variant seems to be in use in Ontario/Quebec vice the unadorned one which is used elsewhere.

This would produce a network that looks like this:

![image](https://user-images.githubusercontent.com/3254090/200141013-222dcb71-d92f-4b15-a7a1-cf2a63007494.png)

However, all of the green are trunk roads, and I suspect there are no trunks of any kind being rendered at z4. So I concur with Clay's assessment here that we'd have to change the OMT side.

**Comment by andrepoiy** [comment]
The Trans-Canada Highway would certainly make a good starting point, however, there are a lot of other routes that are arguably more important than the TCH that should also get the render. 

For example, much of the Ontario and Quebec freeway network is not a part of the TCH. In fact, Highway 401, which services the Windsor-Quebec City corridor (the Canadian equivalent to the US Northeast Megalopolis) is not a part of the TCH at any point.

In BC, the TCH between Vancouver and Kamloops is actually not the preferred route, as it takes a longer route and meanders a lot. The preferred route here is Highway 5, which takes a shorter route and is also a dual-carriageway. 

Halifax, NS, the largest city in Atlantic Canada, is also unconnected by the TCH

Other examples include a portion of Alberta Highway 2 and a portion of Sask. Highway 11, which connects the largest cities in their respective provinces but also not a part of the TCH

**Comment by andrepoiy** [comment]
> It's a valid issue, and I'd like to see route networks that are roughly equivalent to US interstates in other countries render at the same zoom level. I haven't tried to tackle this because I'm far less familiar with other countries' route networks.
> 
> One answer would be to promote all of `network=CA:transcanada` and `network=CA:transcanada:namedRoute` to be at a level equivalent to US interstates. I'm not sure what the difference is between the two network values other than the `namedRoute` variant seems to be in use in Ontario/Quebec vice the unadorned one which is used elsewhere.
> 

The named:route variants used in Quebec and Ontario are essentially alternate routes. Some of them make no sense at all for actual cross-country traffic such as the Southern Ontario Route.

Ontario and Quebec don't take the TCH that seriously. In fact, those provinces do not sign the TCH at junctions at all.

**Comment by ZeLonewolf** [comment]
> there are a lot of other routes that are arguably more important than the TCH that should also get the render.

I'm confident that rendering all trunk roads at z4 would be a non-starter on the OpenMapTiles side. Ultimately there needs to be some tagging-based mechanism to determine _which_ roads are the most important to write the code to promote them.

> The named:route variants used in Quebec and Ontario are essentially alternate routes. Some of them make no sense at all for actual cross-country traffic such as the Southern Ontario Route.

That would certainly make accomplishing the goal quite difficult here. Ultimately computer-based maps have to use rules-based rendering, so I'd welcome an exploration on what the "rule" should be to see a road at the highest zoom.

**Comment by zekefarwell** [comment]
I think for Canada, the rules would have to be set on a Province level.  In Ontario, the [400 series highways](https://en.wikipedia.org/wiki/Autoroutes_of_Quebec) are equivalent to US Interestates in importance.  In Quebec, [the Autoroutes](https://en.wikipedia.org/wiki/Autoroutes_of_Quebec) are at the same level.  Quebec Autoroutes can be identified by `network=CA:QC:A`.  Ontario 400 series highways unfortunately do not have a distinct network value so they would have to be identified by `network=CA:ON:primary` and `ref == 400 - 499`.  I'm not familiar enough with the other provinces to know which would be the equivalent of Interstate level importance there.

**Comment by 1ec5** [comment]
> With the completion of [openmaptiles/openmaptiles#1439](https://github.com/openmaptiles/openmaptiles/issues/1439), once this goes live (no known timeline), we will have some major Canadian routes up to z4.

z4 tiles now have these highways. Do we want to show them by decrementing this constant?

https://github.com/osm-americana/openstreetmap-americana/blob/ee0093e86a224c32dfb403dcd0070a43aaf0c0f9/src/layer/road.js#L9

Before:

<img width="3456" height="1934" alt="Nothing in Canada or Mexico. No auxiliary Interstates." src="https://github.com/user-attachments/assets/4d7dbbe6-6b57-42de-b2f6-93de2a03d668" />

After:

<img width="3456" height="1934" alt="Trans-Canada Highway, auxiliary Interstates, and some federal highways in Mexico." src="https://github.com/user-attachments/assets/3b2e62f4-57ae-457d-bd62-dd3fce61f8a1" />

Before:

<img width="3456" height="1934" alt="No highways in Europe." src="https://github.com/user-attachments/assets/54bb6090-a8b8-4eaf-9565-77429d9e1cfa" />

After:

<img width="3456" height="1934" alt="Highways throughout Europe." src="https://github.com/user-attachments/assets/309e8eb5-4dbf-4769-ae0e-dea50d2e982c" />

Before:

<img width="3456" height="1934" alt="No highways in East Asia." src="https://github.com/user-attachments/assets/b088be69-67a8-4cc7-ab9b-4c65752ada09" />

After:

<img width="3456" height="1934" alt="Highways crisscrossing eastern China, Taiwan, South Korea, and Japan." src="https://github.com/user-attachments/assets/3a202d26-2968-49d4-a42d-c5ecd0a112a8" />

**Comment by minchinweb** [comment]
> roughly equivalent to US interstates

This is likely the Canadian [National Highway System](https://en.wikipedia.org/wiki/National_Highway_System_(Canada)?useskin=vector), at least it terms of deemed importance although not the same freeway standards that have been applied to the Interstate.

Here's an [overview map](https://tc.canada.ca/sites/default/files/migrated/nhs_2007.pdf)

<img width="997" height="710" alt="Image" src="https://github.com/user-attachments/assets/594d4182-9109-4767-86d8-09f5e55078d9" />

It seems it's already [tagged in OpenStreetMap](https://wiki.openstreetmap.org/wiki/Key:NHS), generally as `NHS=yes` (but also `NHS=core`, `NHS=Feeder`, and `NHS=remote`).

**Comment by ZeLonewolf** [comment]
I suspect that openmaptiles/openmaptiles#1767 will provide a partial fix.  I'm doing some testing now, but some Canadian routes that are supposed to be showing up aren't.

**Comment by 1ec5** [comment]
This issue is about the fact that nothing shows up in Canada at z4. openmaptiles/openmaptiles#1439 addressed that on the server side, modulo openmaptiles/openmaptiles#1767, but we would still need to adjust the filter on the client side. The question is whether we want to open the floodgates to trunk roads at z4. It would be a noticeable worldwide change, because OpenMapTiles doesn’t otherwise filter out trunk roads at that zoom level.

**Comment by ZeLonewolf** [comment]
The server-side behavior in https://github.com/openmaptiles/openmaptiles/issues/1439 is currently non-functional, but will be fixed by https://github.com/openmaptiles/openmaptiles/pull/1767

Since z4 motorways show up in the US, I expect they would also show up in Canada once they're in the tiles.
